pkg/operator/status.go

```diff
@@ -148,22 +148,33 @@ func (optr *Operator) syncDegradedStatus(ierr error) (err error) {
 	optrVersion, _ := optr.vStore.Get("operator")
 	degraded := configv1.ConditionTrue
 	var message, reason string
-	if ierr == nil {
+	if ierr.error == nil {
 		degraded = configv1.ConditionFalse
+		reason = ""
 	} else {
 		if optr.vStore.Equal(co.Status.Versions) {
 			// syncing the state to exiting version.
 			message = fmt.Sprintf("Failed to resync %s because: %v", optrVersion, ierr.Error())
 		} else {
 			message = fmt.Sprintf("Unable to apply %s: %v", optrVersion, ierr.Error())
 		}
-		reason = failedToSyncReason
+		reason = ierr.task + "SyncError"
 
 		// set progressing
 		if cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorProgressing) {
-			cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorProgressing, Status: configv1.ConditionTrue, Message: fmt.Sprintf("Unable to apply %s", optrVersion)})
+			cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorProgressing,
+				Status:  configv1.ConditionTrue,
+				Message: message,
+				Reason:  reason,
+			})
 		} else {
-			cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorProgressing, Status: configv1.ConditionFalse, Message: fmt.Sprintf("Error while reconciling %s", optrVersion)})
+			cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorProgressing,
+				Status:  configv1.ConditionFalse,
+				Message: message,
+				Reason:  reason,
```

Comment:
wondering if it's right to report the Degraded message and reason here for progressing as well

Reply:
I'd avoid this change if not strictly required for the linked BZ, we can come back to this later on

Reply:
we set the message here in the past. I replaced `fmt.Sprintf("Error while reconciling %s", optrVersion)}` with the message set above. Want me to remove message and reason here entirely?

Reply:
nope, just rollback to what we had, so that this PR has a clear scope on the BZ